code/unclassified/src/tokenizer/tokenizer.cpp

```diff
@@ -48,5 +55,7 @@ int main()
     ans = myStrTok(nullptr, ' ');
     cout << endl << ans << endl;
 
+    delete ans;
```

Comment:
See, the usage of `std::string` would make this easier.

Reply:
@christianbender I'm just saying that you really don't need pointers for tokenization. You might have to rewrite the algorithm, but if you don't want to, you can leave this as it is.

Reply:
@arnavb The problem is the function must return a string. But this must be a pointer of type ```std::string*``` .  I must use the heap because the function tidy up its stack.